### PR TITLE
feat(gossipsub): deprecate `Config::idle_timeout`

### PR DESCRIPTION
**Pull request by leonzchang**
## Description

Deprecate the `Config::idle_timeout` function in preparation for removing the `KeepAlive::Until` entirely.

Related: #3844.

## Change checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Review by thomaseizinger — APPROVED** [review]
Great, thank you!

**Comment by thomaseizinger** [comment]
It seems like you'll need to place a few select `#[allow(deprecated)]` in the code for CI to pass!

**Comment by leonzchang** [comment]
> It seems like you'll need to place a few select `#[allow(deprecated)]` in the code for CI to pass!

Thanks, added 7ff11f3

File: protocols/gossipsub/src/config.rs [hunk 1]
```diff
@@ -186,6 +186,10 @@ impl Config {
     /// The time a connection is maintained to a peer without being in the mesh and without
     /// send/receiving a message from. Connections that idle beyond this timeout are disconnected.
     /// Default is 120 seconds.
+    #[deprecated(
+        note = "Set a global idle connection timeout via `SwarmBuilder::idle_connection_timeout` instead."
+    )]
+    #[allow(deprecated)]
```
Comment (thomaseizinger) [inline protocols/gossipsub/src/config.rs]: You'll have to put that on the _usage_ site of the method. See the annotations below :)

Comment (leonzchang) [reply protocols/gossipsub/src/config.rs]: Oops, My bad! I thought was placing it on the deprecated function. 😅
Fixed 8b565ea

**Pull request merged**